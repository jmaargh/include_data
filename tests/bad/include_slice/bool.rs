fn main() {}

// Non-Pod type
static BYTES_32_bool: &[bool] = include_data::include_slice!(bool, "../../test_data/binary_32");
