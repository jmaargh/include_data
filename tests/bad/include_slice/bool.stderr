error[E0277]: the trait bound `bool: bytemuck::pod::Pod` is not satisfied
 --> tests/bad/include_slice/bool.rs:4:33
  |
4 | static BYTES_32_bool: &[bool] = include_data::include_slice!(bool, "../../test_data/binary_32");
  |                                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `bytemuck::pod::Pod` is not implemented for `bool`
  |
  = help: the following other types implement trait `bytemuck::pod::Pod`:
            ()
            ManuallyDrop<T>
            Option<T>
            PhantomData<T>
            PhantomPinned
            Wrapping<T>
            [T; N]
            __m128
          and $N others
note: required by a bound in `AlignedAs`
 --> src/lib.rs
  |
  | pub struct AlignedAs<T: bytemuck::Pod, B: Bytes + ?Sized> {
  |                         ^^^^^^^^^^^^^ required by this bound in `AlignedAs`
  = note: this error originates in the macro `include_data::include_slice` (in Nightly builds, run with -Z macro-backtrace for more info)
